app/js/actions/connection.js

```diff
@@ -11,6 +17,51 @@ const establishAvailableAPIList = (availableApis) => ({
     availableApis
 });
 
+const alertReducerOfConnection = () => ({
+    type: 'HANDSHAKE_SERVER'
+});
+
+export const successfullyConnected = (result) => ({
+    type: 'SUCCESFULLY_CONNECTED',
+    result
+});
+
+const updateConnectionStatus = (status) => ({
+    type: 'UPDATE_STATUS',
+    status
+});
+
+const shakeHandsWithServer = () => {
+    return (dispatch, getState) => {
+        dispatch(alertReducerOfConnection());
+        const { connection: { uri, availableApis, secretKey } } = getState();
+        const byteArray = CryptoJS.enc.Base64.parse(secretKey);
+        const requestTime = Date.now();
+        const message = ['POST', window.location.origin, requestTime, 'application/json', 0];
+        const hmac = CryptoJS.algo.HMAC.create(CryptoJS.algo.SHA256, byteArray);
+        message.map(value => hmac.update(value.toString()));
+        const hash = hmac.finalize().toString(CryptoJS.enc.Base64);
```

Comment:
We will want to support other methods, and we will need to sign each one. E.g. `PUT`, `DELETE`